Lib/test/test_builtin.py

```diff
@@ -1511,6 +1513,126 @@ def test_construct_singletons(self):
             self.assertRaises(TypeError, tp, 1, 2)
             self.assertRaises(TypeError, tp, a=1, b=2)
 
+
+class TestBreakpoint(unittest.TestCase):
+    def test_breakpoint(self):
+        with patch('pdb.set_trace') as mock:
+            breakpoint()
+        mock.assert_called_once()
+
+    def test_breakpoint_with_breakpointhook_set(self):
+        call_status = 'Not called'
+        def my_breakpointhook():
+            nonlocal call_status
+            call_status = 'Called'
+        try:
+            sys.breakpointhook = my_breakpointhook
+            breakpoint()
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
```

Comment:
I suggest to use support.swap_attr(sys, "breakpointhook", ...): it's shorter and restore the variable to its previous value. Again, the test runner might have replaced sys.breakpointhook for good reasons. Don't reset it to sys.__breakpointhook__.

Reply:
Excellent suggestion.  I'm going to use this as mentioned above.